rest/tests/test_rest_authorization.py

```diff
@@ -642,3 +648,234 @@ def test_members_delete(self):
+        self.get_learning_resource(self.repo.slug, lr_id)
+
+        # user_norepo has no view_repo permission
+        self.logout()
+        self.login(self.user_norepo.username)
+        self.get_learning_resources(
+            self.repo.slug, expected_status=HTTP_403_FORBIDDEN)
+        self.get_learning_resource(
+            self.repo.slug, lr_id, expected_status=HTTP_403_FORBIDDEN)
+
+        # as anonymous
+        self.logout()
+        self.get_learning_resources(
+            self.repo.slug, expected_status=HTTP_403_FORBIDDEN)
+        self.get_learning_resource(
+            self.repo.slug, lr_id, expected_status=HTTP_403_FORBIDDEN)
+
+    def test_resource_create(self):
+        """Test create learning resource"""
+        resp = self.client.post(
+            '{repo_base}{repo_slug}/learning_resources/'.format(
+                repo_base=REPO_BASE,
+                repo_slug=self.repo.slug,
+            ),
+            self.DEFAULT_LR_DICT
+        )
+        self.assertEqual(resp.status_code, HTTP_405_METHOD_NOT_ALLOWED)
+
+    def test_resource_delete(self):
+        """Test delete learning resource"""
+        tarball_file = self.get_course_single_tarball()
+        import_file(
+            tarball_file, self.repo.id, self.user.id)
+        resource = get_resources(self.repo.id).first()
+        resource.static_assets.add(StaticAsset.objects.first())
+        lr_id = self.get_learning_resources(self.repo.slug)['results'][0]['id']
+
+        resp = self.client.delete(
+            '{repo_base}{repo_slug}/learning_resources/{lr_id}/'.format(
+                repo_base=REPO_BASE,
+                repo_slug=self.repo.slug,
+                lr_id=lr_id,
+            ),
+            self.DEFAULT_LR_DICT
+        )
+        self.assertEqual(resp.status_code, HTTP_405_METHOD_NOT_ALLOWED)
+
+    def test_resource_put_patch(self):
+        """Test update learning resource"""
+        tarball_file = self.get_course_single_tarball()
+        import_file(
+            tarball_file, self.repo.id, self.user.id)
+        resource = get_resources(self.repo.id).first()
+        resource.static_assets.add(StaticAsset.objects.first())
+        lr_id = self.get_learning_resources(self.repo.slug)['results'][0]['id']
+
+        new_description_dict = {"description": "new description"}
+
+        # as curator who has add_edit_metadata permission
+        self.logout()
+        self.login(self.curator_user.username)
+        self.patch_learning_resource(
+            self.repo.slug, lr_id, new_description_dict
+        )
+        self.put_learning_resource(
+            self.repo.slug, lr_id, new_description_dict
+        )
+
+        # author does have add_edit_metadata permission
+        self.logout()
+        self.login(self.author_user.username)
+        self.patch_learning_resource(
+            self.repo.slug, lr_id, new_description_dict
+        )
+        self.put_learning_resource(
```

Comment:
The functions assert that for every key present in the input dictionary, the output value matches up with the input value. I'll make sure the input data is updated before each call to properly test a change in state